src/state/GameState.ts

```diff
@@ -312,6 +312,22 @@ export class GameState {
         return this.players.filter((player) => player.teamName === teamName);
     }
 
+    public getMaxActivePlayers(): number {
+        let maxActivePlayers = 0;
```

Comment:
We can't determine the maximum number of players who can play at once based on this logic. If both teams start with 2 players, and another player joins the first team, then we wouldn't allow it because the maximum number of active players was 2.

The game format determines how many active players are allowed (for example, 4 in ACF, 5 in Illinois Scholastic Bowl). We have to add more information into the game formats through `IGameFormat` to specify what the maximum allowed is, and then drive our logic based on that. The field would be based on this one from the QB Schema: https://github.com/quizbowl/schema/blob/503823451cbb0802dfc56982b5909e7779f9b965/schema/tournament.graphql#L123

That change is a little more involved. It requires:
- Adding a field for `maximumActivePlayers` in `IGameFormat`
- Adding those fields in the existing game formats defined in `GameFormats.ts`
- Add logic to `getUpgradedFormatVersion` to fill in this value for old games (could use `4` as a default value)
- Add some UI to update this value in `CustomizeGameFormatDialog`
- Add some tests in `GameFormatTests.ts`
- Add the remaining code in this PR

Reply:
We don't need a specific format for IHSA (unless someone wants to write one), but allowing a reasonable value is enough. This still requires having fields for it in `IGameFormat` so that readers can override the value.